src/test_vector.rs

```diff
@@ -0,0 +1,229 @@
+// SPDX-License-Identifier: MPL-2.0
+
+//! Module `test_vector` generates test vectors of serialized Prio inputs and
+//! support for working with test vectors, enabling backward compatibility
+//! testing.
+
+use crate::{
+    client::{Client, ClientError},
+    encrypt::{PrivateKey, PublicKey},
+    field::FieldElement,
+    server::{Server, ServerError},
+};
+use rand::Rng;
+use serde::{Deserialize, Serialize};
+use std::{convert::TryFrom, fmt::Debug};
+
+/// Errors propagated by functions in this module.
+#[derive(Debug, thiserror::Error)]
+pub enum TestVectorError {
+    /// Error from Prio client
+    #[error("Prio client error {0}")]
+    Client(#[from] ClientError),
+    /// Error from Prio server
+    #[error("Prio server error {0}")]
+    Server(#[from] ServerError),
+    /// Error while converting primitive to FieldElement associated integer type
+    #[error("Integer conversion error {0}")]
+    IntegerConversion(String),
+}
+
+const SERVER_1_PRIVATE_KEY: &str =
+    "BIl6j+J6dYttxALdjISDv6ZI4/VWVEhUzaS05LgrsfswmbLOgNt9HUC2E0w+9RqZx3XMkdEHBH\
+     fNuCSMpOwofVSq3TfyKwn0NrftKisKKVSaTOt5seJ67P5QL4hxgPWvxw==";
+const SERVER_2_PRIVATE_KEY: &str =
+    "BNNOqoU54GPo+1gTPv+hCgA9U2ZCKd76yOMrWa1xTWgeb4LhFLMQIQoRwDVaW64g/WTdcxT4rD\
+     ULoycUNFB60LER6hPEHg/ObBnRPV1rwS3nj9Bj0tbjVPPyL9p8QW8B+w==";
+
+/// An ECDSA P-256 private key suitable for decrypting inputs, used to generate
+/// test vectors and later to decrypt them.
+fn server_1_private_key() -> PrivateKey {
+    PrivateKey::from_base64(SERVER_1_PRIVATE_KEY).unwrap()
+}
+
+/// The public portion of [`server_1_private_key`].
+fn server_1_public_key() -> PublicKey {
+    PublicKey::from(&server_1_private_key())
+}
+
+/// An ECDSA P-256 private key suitable for decrypting inputs, used to generate
+/// test vectors and later to decrypt them.
+fn server_2_private_key() -> PrivateKey {
+    PrivateKey::from_base64(SERVER_2_PRIVATE_KEY).unwrap()
+}
+
+/// The public portion of [`server_2_private_key`].
+fn server_2_public_key() -> PublicKey {
+    PublicKey::from(&server_2_private_key())
+}
+
+/// A test vector of Prio inputs, serialized and encrypted in the Priov2/ENPA
+/// format, along with a reference sum.
+#[derive(Debug, Eq, PartialEq, Serialize, Deserialize)]
+pub struct EnpaTestVector<F> {
+    /// Base64 encoded private key for the "first" a.k.a. "PHA" server, which
+    /// may be used to decrypt `server_1_shares`.
+    pub server_1_private_key: String,
+    /// Base64 encoded private key for the non-"first" a.k.a. "facilitator"
+    /// server, which may be used to decrypt `server_2_shares`.
+    pub server_2_private_key: String,
+    /// Dimension (number of buckets) of the inputs
+    pub dimension: usize,
+    /// Encrypted shares of ENPA format inputs for the "first" a.k.a. "PHA"
+    /// server. The inner `Vec`s are encrypted bytes.
+    pub server_1_shares: Vec<Vec<u8>>,
```

Comment:
- [ ] `serde_json` serialization of a `Vec[u8]` is terrible; see if we can get it to serialize to a base64 string instead